### PR TITLE
build: add coverage check

### PR DESCRIPTION
**Pull request by baixiaokuang**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -375,7 +406,23 @@ if(ENABLE_LCAO)
...
-    TYPE BIN
-    #DESTINATION ${CMAKE_INSTALL_BINDIR}
-)
+if(DEFINED ENV{ABACUSROOT} AND NOT DEFINED ABACUSROOT)
+  set(ABACUSROOT "$ENV{ABACUSROOT}")
+endif()
+if(DEFINED ABACUSROOT)
+  install(PROGRAMS ${ABACUS_BIN_PATH}
+      TYPE BIN
+      #DESTINATION ${ABACUSROOT}
```
Comment (caic99) [inline CMakeLists.txt]: Looks like `${ABACUSROOT}` is not in use?

File: CMakeLists.txt [hunk 2]
```diff
@@ -325,6 +351,11 @@ IF (BUILD_TESTING)
   function(AddTest) # function for UT
     cmake_parse_arguments(UT "DYN" "TARGET" "LIBS;DYN_LIBS;STATIC_LIBS;SOURCES;DEPENDS" ${ARGN})
     add_executable(${UT_TARGET} ${UT_SOURCES})
+
+    if(ENABLE_COVERAGE)
+      add_coverage(${UT_TARGET})
```
Comment (caic99) [inline CMakeLists.txt:356]: Just to clarify, is `add_coverage` defined in `cmakecodecov`?

Comment (caic99) [inline CMakeLists.txt:356]: Can we add compilation flags manually, preventing `add_coverage` for each target?

**Review by caic99 — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (baixiaokuang) [reply CMakeLists.txt:356]: Yes

Comment (baixiaokuang) [reply CMakeLists.txt:356]: Yes, but dependency error may occur easily, so we may add it later when we have dealt with coverage dependencies.

File: CMakeLists.txt (hunk 1 above)
Comment (baixiaokuang) [reply CMakeLists.txt]: Yes, already removed.

**Pull request merged**